src/System.Device.Gpio/System/Device/Gpio/LibgpiodDriverEventHandler.cs

```diff
@@ -51,7 +51,7 @@ private Task InitializeEventDetectionTask(CancellationToken token, SafeLineHandl
                     TimeSpec timeout = new TimeSpec
                     {
                         TvSec = new IntPtr(0),
-                        TvNsec = new IntPtr(1000000)
+                        TvNsec = new IntPtr(50_000_000)
```

Comment:
Aside, if these aren't pointers and you're compiling with C# 9.0 perhaps it would be cleaner to use `nint` or `nuint`

Reply:
Please understand I just want to fix a big issue for smaller devices. Your comment is about refactoring the code which was not in my scope. Maybe there are even more places where IntPtr is used....THis optimization should happen seperately...

Reply:
TvNsec should be a long, I do agree we should do a pass on the whole repo for cases where the new native types can be used, and we should do that in a separate PR.

Reply:
[Triage] @Alex-111, 1 ms maybe to small, 50 ms maybe to high, can you give us a rational and where you picked that number? 
Any comparison with different values? Is this 50 the optimal one?
What is the moment where the CPU usage is really increasing?
Note: for the comments on the type for TimeSpec, we will address that in a different PR, there maybe other adjustments elsewhere.
And thanks again for raising this PR.

Reply:
My understanding is that the timout time is the time how long the waiting thread blocks. In case of an event the thread should be released anyway regardless the timout time, so even higher values should not be problematic.

The problem of high CPU usage grows when the number of monitored pins increases, so for a typical emebdded linux board I've found out that 50ms is a good value. 
But I also have to mention that I'm not a linux API expert. I'm not sure if increasing the value could have side effects for edge cases.
For me  the change dropped CPU usage to about 0-1%, instead of 12% for one single pin!

Reply:
Sounds reasonable. The only downside of making the delay too long is that the termination will take extra time. But unless you register-deregister-register an event all the time (which is probably a rare scenario) 50ms shouldn't really hurt.